src/app-layer-ssl.c

```diff
@@ -579,6 +588,8 @@ static int TlsDecodeHSCertificate(SSLState *ssl_state, SSLStateConnp *connp,
         TlsDecodeHSCertificateErrSetEvent(ssl_state, err_code);
     if (x509 != NULL)
         rs_x509_free(x509);
+    if (connp->cert0_sans != NULL)
+        SCFree(connp->cert0_sans);
```

Comment:
Should we not `rs_cstring_free(ssl_state->server_connp.cert0_sans[I])` as well ?

Reply:
I thought about it too. But, then the others were not handled here either. e.g. `issuer` and ASAN didn't flag it so, I thought maybe some FFI magic takes care of it?

Reply:
Maybe it is safe today, but it is not sure for tomorrow, so I would do one unique SansFree function that is called from the different places

Reply:
ok. will add